packages/firestore/src/api/database.ts

```diff
@@ -1817,13 +1808,35 @@ export class QuerySnapshot implements firestore.QuerySnapshot {
     return new Query(this._originalQuery, this._firestore);
   }
 
-  get docChanges(): firestore.DocumentChange[] {
-    if (!this._cachedChanges) {
+  docChanges(
+    options?: firestore.SnapshotListenOptions
+  ): firestore.DocumentChange[] {
```

Comment:
I tried to make this work earlier but quickly ran into some issues:
- Symbol.iterator cannot actually be redefined.
- I didn't want to override .length since someone could actually be using it with the new docChanges() method. Granted, this is probably very unlikely.

I played around with this some more though and noticed that when someone writes `querySnap.docChanges`, it actually executes the method but without any `this` context. Hence, I have added an error check in the method that checks for `this === undefined`.

Reply:
BTW, I just pushed a commit that added the changes you suggested. In my original attempt, I was calling the `defineProperty` code multiple times, which ended up breaking. The `this === undefined` seems to work in chai (`expect(snap.docChanges)` calls the method without context), but otherwise you are 100% correct (which is much more than 74%!).